x-pack/metricbeat/module/azure/monitor_service.go

```diff
@@ -262,6 +294,84 @@ func (service *MonitorService) GetMetricDefinitionsWithRetry(resourceId string,
 	return metricDefinitionCollection, nil
 }
 
+func (service *MonitorService) QueryResources(
+	resourceIDs []string,
+	subscriptionID string,
+	namespace string,
+	timegrain string,
+	startTime string,
+	endTime string,
+	metricNames []string,
+	aggregations string,
+	filter string,
+	location string) ([]azmetrics.MetricData, error) {
+
+	var tg *string
+	//var interval string
+
+	if timegrain != "" {
+		tg = &timegrain
+	}
+
+	// API fails with bad request if filter value is sent empty.
+	var metricsFilter *string
+	var top int32
+
+	if filter != "" {
+		metricsFilter = &filter
+		top = int32(10)
```

Comment:
I don't remember why we picked 10 here.

Reply:
That is a mistake. I will set it to nil like in GetMetricValues

Reply:
Wait, I remember we had a conversation with MSFT people around this param: 

https://github.com/Azure/azure-sdk-for-go/issues/22757